frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/PnpmInstaller.java

```diff
@@ -169,42 +173,72 @@ private void installPnpm() throws InstallationException {
-            if (pnpmJsExecutable.exists()) {
-                this.logger.info("No pnpm executable found, creating symlink to {}", pnpmJsExecutable.toPath());
-                try {
-                    Files.createSymbolicLink(pnpmExecutable.toPath(), pnpmJsExecutable.toPath());
-                } catch (IOException e) {
-                    throw new InstallationException("Could not copy pnpm", e);
-                }
-            }
+
+        NodeExecutorConfig executorConfig = new InstallNodeExecutorConfig(this.config);
+        File pnpmJsExecutable = executorConfig.getPnpmCjsPath();
+
+        if (!pnpmJsExecutable.exists()) {
+            throw new InstallationException("Could not link to pnpm executable, no pnpm installation found.");
+        }
+
+        this.logger.info("No pnpm executable found, creating symbolic link to {}.", pnpmJsExecutable.toPath());
+
+        try {
+            Files.createSymbolicLink(pnpmExecutable.toPath(), pnpmJsExecutable.toPath());
+        } catch (IOException e) {
+            throw new InstallationException("Could not create symbolic link for pnpm executable.", e);
+        }
+    }
+
+    private void linkExecutableWindows() throws InstallationException{
+        File nodeInstallDirectory = getNodeInstallDirectory();
+        File pnpmExecutable = new File(nodeInstallDirectory, "pnpm.bat");
+
+        if (pnpmExecutable.exists()) {
+            this.logger.info("Existing pnpm executable found, skipping linking.");
+            return;
+        }
+
+        NodeExecutorConfig executorConfig = new InstallNodeExecutorConfig(this.config);
+        File pnpmJsExecutable = executorConfig.getPnpmCjsPath();
+
+        if (!pnpmJsExecutable.exists()) {
+            throw new InstallationException("Could not link to pnpm executable, no pnpm installation found.");
+        }
+
+        this.logger.info("No pnpm executable found, creating proxy script to {}.", pnpmJsExecutable.toPath());
+
+        Path nodePath = executorConfig.getNodePath().toPath();
+        Path relativeNodePath = nodeInstallDirectory.toPath().relativize(nodePath);
+        Path relativePnpmPath = nodeInstallDirectory.toPath().relativize(pnpmJsExecutable.toPath());
+
+        // Create a batch script that will proxy any commands passed to the pnpm executable.
+        String scriptContents = new StringBuilder()
+                .append(":: Created by frontend-maven-plugin, please don't edit manually.\r\n")
+                .append("@ECHO OFF\r\n")
+                .append("\r\n")
+                .append("SETLOCAL\r\n")
+                .append("\r\n")
+                .append(String.format("SET \"NODE_EXE=%%~dp0\\%s\"\r\n", relativeNodePath))
+                .append(String.format("SET \"PNPM_CLI_JS=%%~dp0\\%s\"\r\n", relativePnpmPath))
+                .append("\r\n")
+                .append("\"%NODE_EXE%\" \"%PNPM_CLI_JS%\" %*")
+                .toString();
```

Comment:
This script is partially taken from the [NPM equivalent](https://github.com/npm/cli/blob/c422a01e1564d25148c821ee257196ebe60e8e6c/bin/npm.cmd), but pruned down to the bare essentials needed for this use-case.